full-moon/src/ast/compound.rs

```diff
@@ -0,0 +1,140 @@
+use crate::ast::{Expression, Var};
+use crate::tokenizer::{Symbol, TokenReference};
+use derive_more::Display;
+use full_moon_derive::{Node, Visit};
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, Display, PartialEq, Eq, Node, Visit)]
+#[cfg_attr(feature = "serde", derive(Deserialize, Serialize))]
+#[non_exhaustive]
+#[allow(missing_docs)]
+#[display("{_0}")]
+/// Compound operators, such as X += Y or X -= Y
+pub enum CompoundOp {
+    PlusEqual(TokenReference),
+    MinusEqual(TokenReference),
+    StarEqual(TokenReference),
+    SlashEqual(TokenReference),
+    CaretEqual(TokenReference),
+
+    // luau sepcific
```

Comment:
Should these be gated via `cfg(feature = "luau")` flags if they are only available in luau?

Some with the below for CfxLua